executor/encoder_pl.py

```
@@ -0,0 +1,127 @@
+        'model_name': 'pointnet',
+        'hidden_dim': 1024,
+        'embed_dim': 1024,
+        'model_path': '',
+    },
+    'PointConv-Shapenet-d1024': {
+        'model_name': 'pointconv',
+        'hidden_dim': 1024,
+        'embed_dim': 1024,
+        'model_path': 'https://jina-pretrained-models.s3.us-west-1.amazonaws.com/mesh_models/pointconv-shapenet-d1024.pth',
+    },
+    'PointNet-Shapenet-d512': {
+        'model_name': 'pointnet',
+        'hidden_dim': 1024,
+        'embed_dim': 512,
+        'model_path': '',
+    },
+    'PointConv-Shapenet-d512': {
+        'model_name': 'pointconv',
+        'hidden_dim': 1024,
+        'embed_dim': 512,
+        'model_path': 'https://jina-pretrained-models.s3.us-west-1.amazonaws.com/mesh_models/pointconv-shapenet-d512.pth',
+    },
+}
+
+
+class MeshDataEncoderPL(pl.LightningModule):
+    def __init__(
+        self,
+        pretrained_model: str = None,
+        default_model_name='pointconv',
+        model_path: Optional[str] = None,
+        hidden_dim: int = 1024,
+        embed_dim: int = 1024,
+        input_shape: str = 'bnc',
+        device: str = 'cpu',
+        batch_size: int = 64,
+        filters: Optional[dict] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.save_hyperparameters()
+
+        model_path = None
+        if pretrained_model in AVAILABLE_MODELS:
+            config = AVAILABLE_MODELS[pretrained_model]
+            model_name = config.pop('model_name')
+            model_path = config.pop('model_path')
+            embed_dim = config.pop('embed_dim')
+            hidden_dim = config.pop('hidden_dim')
+        else:
+            model_name = default_model_name
+        self._model = MeshDataModel(
+            model_name=model_name,
+            hidden_dim=hidden_dim,
+            embed_dim=embed_dim,
+            pretrained=True if model_path else False,
+            input_shape=input_shape,
+        )
+
+        if model_path:
```

what if `model_path` is `None`? Should we raise any exceptions to users?

---

if `model_path` is `None`, self._model is just an initialized MeshDataModel instance without downloading the pretrained model from the https:// network.